htdocs/mrp/mo_production.php

```diff
@@ -890,116 +906,186 @@
+								});
+							});';
+							print '</script>';
+							if (empty($conf->use_javascript_ajax)) {
+								print '<a href="' . $_SERVER["PHP_SELF"] . '?collapse=' . $collapse . ',' . $line->id . '">';
+							}
+							print img_picto($langs->trans("ShowDetails"), "chevron-down", 'id="expandtoproduce' . $line->id . '"');
+							if (empty($conf->use_javascript_ajax)) {
+								print '</a>';
+							}
+						} else {
+							if ($nblinetoconsume == $nblinetoconsumecursor) {    // If it is the last line
+								print '<script>jQuery("#tablelines").removeClass("nobottom");</script>';
+							}
+						}
+						print ' ' . price2num($alreadyconsumed, 'MS');
+						print '</td>';
+						// Warehouse
+						print '<td>';
+						if (!empty($conf->global->STOCK_CONSUMPTION_FROM_MANUFACTURING_WAREHOUSE) && $tmpwarehouse->id > 0) {
+							print img_picto('', $tmpwarehouse->picto) . " " . $tmpwarehouse->label;
+						}
+						print '</td>';
+						// Stock
+						if (isModEnabled('stock')) {
+							print '<td class="nowraponall right">';
+							if (empty($conf->global->STOCK_SUPPORTS_SERVICES) && $tmpproduct->type != Product::TYPE_SERVICE) {
+								if (!$line->disable_stock_change && $tmpproduct->stock_reel < ($line->qty - $alreadyconsumed)) {
+									print img_warning($langs->trans('StockTooLow')) . ' ';
+								}
+								if (empty($conf->global->STOCK_CONSUMPTION_FROM_MANUFACTURING_WAREHOUSE) || empty($tmpwarehouse->id)) {
+									print price2num($tmpproduct->stock_reel, 'MS'); // Available
 								} else {
-									print "0";
+									// Print only the stock in the selected warehouse
+									$tmpproduct->load_stock();
+									$wh_stock = $tmpproduct->stock_warehouse[$tmpwarehouse->id];
+									if (!empty($wh_stock)) {
+										print price2num($wh_stock->real, 'MS');
+									} else {
+										print "0";
+									}
 								}
 							}
+							print '</td>';
+						}
+						// Lot
+						if (isModEnabled('productbatch')) {
+							print '<td></td>';
 						}
-						print '</td>';
-					}
-					// Lot
-					if (isModEnabled('productbatch')) {
-						print '<td></td>';
-					}
 
-					// Split
-					print '<td></td>';
+						// Split
+						print '<td></td>';
 
-					// Split All
-					print '<td></td>';
+						// Split All
+						print '<td></td>';
 
-					// Action delete line
-					if ($permissiontodelete) {
-						$href = $_SERVER["PHP_SELF"].'?id='.((int) $object->id).'&action=deleteline&token='.newToken().'&lineid='.((int) $line->id);
-						print '<td class="center">';
-						print '<a class="reposition" href="'.$href.'">';
-						print img_picto($langs->trans('TooltipDeleteAndRevertStockMovement'), 'delete');
-						print '</a>';
-						print '</td>';
-					}
+						// Action Edit line
+						if ($object->status == Mo::STATUS_DRAFT) {
+							$href = $_SERVER["PHP_SELF"] . '?id=' . ((int) $object->id) . '&action=editline&token=' . newToken() . '&lineid=' . ((int) $line->id);
+							print '<td class="center">';
```

Comment:
You add a new column with <td></td> section here.
but later for exemple into the loop line 1090, you don't have the new column added meaning that there is a bad balance of nb of column between lines.
To match Dolibarr behaviour, it is recommended to consume only 1 line for the column of edit and delete. So can you add the action to edit into the same <td></td> than the delete (at end of line) ? This will solve the bug of wrong nb of columns and standardize code with other tables.